### PR TITLE
Updated create marriage service

### PR DESCRIPTION
**Pull request by smisidjan**

### CODE REVIEW

File: Service/CreateMarriageService.php [hunk 1]
```diff
@@ -59,22 +65,28 @@ class CreateMarriageService
...
+     * @param LoggerInterface        $logger                 The Logger Interface
      */
     public function __construct(
         EntityManagerInterface $entityManager,
+        CacheService $cacheService,
         HandleAssentService $handleAssentService,
         UpdateChecklistService $updateChecklistService,
-        Security $security
+        Security $security,
+        LoggerInterface $logger
```
Comment (rjzondervan) [inline Service/CreateMarriageService.php:80]: If you rename $logger here to $pluginLogger logs are written nicely to the plugin-channel.

File: Service/HandleAssentService.php [hunk 2]
```diff
@@ -56,15 +57,18 @@ class HandleAssentService
...
         EntityManagerInterface $entityManager,
         EventDispatcherInterface $eventDispatcher,
-        MessageBirdService $messageBirdService
+        MessageBirdService $messageBirdService,
+        LoggerInterface $logger
     ) {
         $this->entityManager = $entityManager;
         $this->eventDispatcher = $eventDispatcher;
         $this->messageBirdService = $messageBirdService;
+        $this->logger = $logger;
```
Comment (rjzondervan) [inline Service/HandleAssentService.php:71]: Same as previous comment.

File: Service/CreateMarriageService.php [hunk 3]
```diff
@@ -149,54 +160,90 @@ private function validateCeremonie(array $huwelijk)
...
      * This function creates a person object for the given user.
      */
-    private function createPerson(): ?ObjectEntity
+    private function createPerson(array $huwelijk, ?ObjectEntity $brpPerson = null): ?ObjectEntity
     {
         $personSchema = $this->getSchema('https://klantenBundle.commonground.nu/klant.klant.schema.json');
 
-        // @TODO BRP person has to be set to the klantObject
-        // @TODO get user/ person from jwt token and create a person object
+        if ($brpPerson) {
```
Comment (WilcoLouwerse) [inline Service/CreateMarriageService.php:191]: if ($brpPerson !== null)
or
if ($brpPerson instanceof ObjectEntity === true)

(for codacy and) better readability of your code

File: Service/CreateMarriageService.php [hunk 4]
```diff
@@ -149,54 +160,90 @@ private function validateCeremonie(array $huwelijk)
...
+                $email = $huwelijkPerson['emails'][0]['email'];
+            }
+
+            if (key_exists('telefoonnummers', $huwelijkPerson)) {
+                $phonenumber = $huwelijkPerson['telefoonnummers'][0]['telefoonnummer'];
+            }
+        }
+
         $person = new ObjectEntity($personSchema);
         $person->hydrate([
```
Comment (WilcoLouwerse) [inline Service/CreateMarriageService.php:211]: Just checking, but this is not possible to do with mapping right now?

File: Service/InviteWitnessService.php [hunk 5]
```diff
@@ -120,15 +124,16 @@ private function inviteWitness(array $huwelijk, ?string $id): ?array
...
 
             throw new GatewayException('Could not find huwelijk with id '.$id);
         }
 
-        if (isset($huwelijk['getuigen'])) {
-
-            // @TODO check if witness is aldready set or overwrite the witness array
+        if (isset($huwelijk['getuigen']) && count($huwelijk['getuigen']) <= 4) {
+            // @TODO overwrite the witness array in the huwelijkObject
+            // @TODO check if witness is aldready set
```
Comment (WilcoLouwerse) [inline Service/InviteWitnessService.php:136]: Todo, maybe do this and / or remove todo if possible

File: Service/InviteWitnessService.php [hunk 6]
```diff
@@ -142,6 +147,9 @@ private function inviteWitness(array $huwelijk, ?string $id): ?array
 
             $huwelijkObject->setValue('getuigen', $witnessAssents);
 
+            // @TODO update checklist with getuigen
```
Comment (WilcoLouwerse) [inline Service/InviteWitnessService.php:150]: Todo, maybe do this and / or remove todo if possible

**Review by rjzondervan — APPROVED** [review]
Overall good work, if you rename the logger-variables in the constructor (the name of the global variable does not matter), you can decide the channel the logs are written in.

**Review by WilcoLouwerse — APPROVED** [review]

**Pull request merged**